### PR TITLE
Implement Cucumber test for Student uploading an offer letter

### LINKED ISSUE #67: Implement Cucumber test for uploading an offer letter (Student)

### PR DESCRIPTION
**Pull request by kraglalbert**
# Summary

Implemented the Cucumber test for a Student uploading their offer letter in `StudentSubmitOfferLetterIT`

## Test Plan

Ran `mvn failsafe:integration-test` and checked to see if `StudentSubmitOfferLetterIT` passed

## Related Issues

closes #67

### CODE REVIEW

File: Backend/src/test/java/ca/mcgill/cooperator/cucumber/StudentSubmitOfferLetterIT.java [hunk 1]
```diff
@@ -1,27 +1,201 @@
...
+
+    @Before
+    public void clearDatabase() {
+        List<CoopDetails> coopDetails = coopDetailsService.getAllCoopDetails();
+        for (CoopDetails cd : coopDetails) {
+            cd.setCoop(null);
+            coopDetailsRepository.save(cd);
+        }
+        // deleting all students will also delete all coops
+        studentRepository.deleteAll();
```
Comment (susanmatuszewski) [inline Backend/src/test/java/ca/mcgill/cooperator/cucumber/StudentSubmitOfferLetterIT.java]: are you sure coop details gets deleted in this clear database?

File: Backend/src/test/java/ca/mcgill/cooperator/cucumber/StudentSubmitOfferLetterIT.java [hunk 2]
```diff
@@ -1,27 +1,201 @@
...
     @When("the Student uploads a copy of their offer letter")
-    public void studentSubmitsOfferLetter() {
-        assertTrue(true);
+    public void studentSubmitsOfferLetter() throws Exception {
+        // set up global variables
+        student = createTestStudent();
+        employerContact = createTestEmployerContact();
+        course = createTestCourse();
+        courseOffering = createTestCourseOffering(course);
+        testCoop = coopService.createCoop(CoopStatus.UNDER_REVIEW, courseOffering, student);
```
[the remark this answers is not in the document]
Comment (susanmatuszewski) [reply Backend/src/test/java/ca/mcgill/cooperator/cucumber/StudentSubmitOfferLetterIT.java]: i have methods written in coop IT that creates all of the related dtos with post requests as well if you want to use those

File: Backend/src/test/java/ca/mcgill/cooperator/cucumber/StudentSubmitOfferLetterIT.java [hunk 3]
```diff
@@ -1,27 +1,224 @@
...
+        // get object from response
+        List<StudentReportDto> returnedReports =
+                Arrays.asList(
+                        objectMapper.readValue(
+                                mvcResult.getResponse().getContentAsString(),
+                                StudentReportDto[].class));
+
+        assertEquals(1, returnedReports.size());
+
+        StudentReportDto report = returnedReports.get(0);
```
Comment (PaulHooley) [inline Backend/src/test/java/ca/mcgill/cooperator/cucumber/StudentSubmitOfferLetterIT.java]: add some loop here? would be less sketchy than get(0)

File: Backend/src/test/java/ca/mcgill/cooperator/cucumber/StudentSubmitOfferLetterIT.java [hunk 4]
```diff
@@ -1,27 +1,224 @@
...
+                                .file("file", multipartFile.getBytes())
+                                .param("status", "IN_REVIEW")
+                                .param("title", "Offer Letter")
+                                .param("coop_id", String.valueOf(testCoop.getId()))
+                                .contentType(MediaType.MULTIPART_FORM_DATA)
+                                .characterEncoding("utf-8"))
+                .andExpect(status().isOk());
     }
 
     @And("submits the details of their Coop term")
```
Comment (PaulHooley) [inline Backend/src/test/java/ca/mcgill/cooperator/cucumber/StudentSubmitOfferLetterIT.java]: Change to CoopDetails

**Pull request merged**